build-release-pkgs.xml

```diff
@@ -115,7 +115,13 @@
     <zip basedir="${workdir}" destfile="${workdir}/narayana-full-${tag}-src.zip" includes="${tag}/**"/>
     <!-- build the binary releases -->
     <exec dir="${workdir}/${tag}" executable="${mvn.executable}" failonerror="true">
-      <arg line="deploy -DskipTests -gs tools/maven/conf/settings.xml -Dorson.jar.location=${user.dir}/ext/ -Prelease"/>
+      <arg line="clean deploy -DskipTests -gs tools/maven/conf/settings.xml -Dorson.jar.location=${user.dir}/ext/ -Prelease"/>
+    </exec>
+    <exec dir="${workdir}/${tag}" executable="${mvn.executable}" failonerror="true">
+      <arg line="clean deploy -DskipTests -gs tools/maven/conf/settings.xml -Prelease -f blacktie/utils/cpp-plugin/pom.xml"/>
+    </exec>
+    <exec dir="${workdir}/${tag}" executable="${mvn.executable}" failonerror="true">
+      <arg line="clean deploy -DskipTests -gs tools/maven/conf/settings.xml -Prelease  -f blacktie/pom.xml -pl :blacktie-jatmibroker-nbf -am"/>
```

Comment:
Thanks for the explanation. I think it's good to merge.

On Mon, Oct 5, 2015 at 4:04 PM, Tom Jenkinson <notifications@github.com>
wrote:

> In build-release-pkgs.xml
> <https://github.com/jbosstm/narayana/pull/924#discussion_r41154365>:
>
> > @@ -115,7 +115,13 @@
> >      <zip basedir="${workdir}" destfile="${workdir}/narayana-full-${tag}-src.zip" includes="${tag}/**"/>
> >      <!-- build the binary releases -->
> >      <exec dir="${workdir}/${tag}" executable="${mvn.executable}" failonerror="true">
> > -      <arg line="deploy -DskipTests -gs tools/maven/conf/settings.xml -Dorson.jar.location=${user.dir}/ext/ -Prelease"/>
> > +      <arg line="clean deploy -DskipTests -gs tools/maven/conf/settings.xml -Dorson.jar.location=${user.dir}/ext/ -Prelease"/>
> > +    </exec>
> > +    <exec dir="${workdir}/${tag}" executable="${mvn.executable}" failonerror="true">
> > +      <arg line="clean deploy -DskipTests -gs tools/maven/conf/settings.xml -Prelease -f blacktie/utils/cpp-plugin/pom.xml"/>
> > +    </exec>
> > +    <exec dir="${workdir}/${tag}" executable="${mvn.executable}" failonerror="true">
> > +      <arg line="clean deploy -DskipTests -gs tools/maven/conf/settings.xml -Prelease  -f blacktie/pom.xml -pl :blacktie-jatmibroker-nbf -am"/>
>
> cpp-plugin lives outside the lifecycle of the main build - it should
> really be in a separate repo tbh but it was a decision we made a while
> back. It basically is a plugin used by the main build to build the C++
> components under maven in a platform neutral manner. You can't actually
> call it directly from the maven build as it is cycling during maven
> initialization.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jbosstm/narayana/pull/924/files#r41154365>.
>